src/core_plugins/kibana/public/dashboard/dashboard_container_api.js

```diff
@@ -0,0 +1,38 @@
+import { ContainerAPI } from 'ui/embeddable';
+
+export class DashboardContainerAPI extends ContainerAPI {
+  constructor(dashboardState, queryManager) {
+    super();
+    this.dashboardState = dashboardState;
+    this.queryManager = queryManager;
+  }
+
+  addFilter(field, value, operator, index) {
+    this.queryManager.add(field, value, operator, index);
+  }
+
+  updatePanel(panelIndex, panelAttributes) {
+    const panelToUpdate = this.dashboardState.getPanels().find((panel) => panel.panelIndex === panelIndex);
+    Object.assign(panelToUpdate, panelAttributes);
+    this.dashboardState.saveState();
+    return panelToUpdate;
+  }
+
+  getAppState() {
+    return this.dashboardState.appState;
+  }
+
+  getIsViewOnlyMode() {
+    return this.dashboardState.getIsViewMode();
+  }
+
+  getInitialState(path, uiState) {
```

Comment:
Haha, I wasn't super clear, I was asking for `createChildUiState(path, initialState)`

This method returns an instance of `PersistedState`, which we usually refer to as `uiState`. This method also takes an argument which it calls `uiState`, but it's actually just a plain object that is used as the initial state for the `uiState` that is returned.